app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,76 @@
+
+class RentalsController < ApplicationController
+  def checkout
+    customer_id = check_params[:customer_id]
+    movie_id = check_params[:movie_id]
+
+    find_customer_and_movie(customer_id, movie_id)
+
+    errors = Rental.check_dependencies(@customer, @movie)
+
+    if errors
+      render json: {errors: errors}, status: :not_found
+    else
+
+      @rental = Rental.create_checkout(@customer, @movie)
+
+      if @rental.save
+
+        Rental.process_checkout(@customer, @movie)
```

Comment:
This dance of checking the dependencies, creating the rentals, and then updating the dependent models is pretty complicated - it might make sense to encapsulate the whole thing in a model method. One thing that might help is using a transaction - go look this up!